common/src/main/java/io/netty/util/ConstantPool.java

```diff
@@ -68,7 +68,7 @@ public T valueOf(String name) {
     private T getOrCreate(String name) {
         T constant = constants.get(name);
         if (constant == null) {
-            final T tempConstant = newConstant(nextId(), name);
+            final T tempConstant = newConstant(0, name);
```

Comment:
`always passing 0 here instead of nextId() seems like a semantic change.`

`nextId()` moved in `newContant()` method so it is doing the same.

` interface appears to depend on some sort of "uniqueness" (as determined by nextId()) for id`

This uniquiness was never used for id, that's why I removed it.